src/main/java/org/jabref/gui/maintable/SearchDialogColumnPreferences.java

```diff
@@ -0,0 +1,24 @@
+package org.jabref.gui.maintable;
+
+import java.util.List;
+
+public class SearchDialogColumnPreferences implements AbstractColumnPreferences {
+
+    private final List<MainTableColumnModel> columns;
+    private final List<MainTableColumnModel> columnSortOrder;
+
+    public SearchDialogColumnPreferences(List<MainTableColumnModel> columns, List<MainTableColumnModel> columnSortOrder) {
+        this.columns = columns;
+        this.columnSortOrder = columnSortOrder;
+    }
+
+    @Override
+    public List<MainTableColumnModel> getColumns() {
+        return columns;
+    }
+
+    @Override
+    public List<MainTableColumnModel> getColumnSortOrder() {
+        return columnSortOrder;
+    }
+}
```

Comment:
except for the two constants in MainTableColumnPreferences this abstraction layer seems to be just code duplication.
Why don't you leave it with the ColumnPreferences?

Reply:
Good point I think this could work. Will test it tomorrow